File: character/player.py
import pyray as p
from raylib import *
import time as t

class Player:
    def __init__(self, posx, posy, width, height, c):
        self.rec = p.Rectangle(posx, posy, width, height)
        self.color = c
        self.isOnFloor = True
        self.currentTime = int(t.time())

        self.ceilingHeight = 30
        self.floorHeight = 1000
        self.frames = 0
        self.ticks = 0
        self.tickTimer = 0
        self.stopIncrementingTickTimer = False
         
        self.xvelocity = 0
        self.yvelocity = 0 

    def Draw(self):
        DrawRectangleRec(self.rec, self.color)

    def Update(self):
        self.frames += 1
        if self.frames % 6 == 0:
            self.ticks += 1
        
        if self.rec.x <= -400:
            self.xvelocity = 0
            self.rec.x = -400

        # collision
        if self.rec.y > self.floorHeight:
            self.rec.y = self.floorHeight

        if self.rec.y < self.ceilingHeight:
            self.yvelocity = 0
            self.rec.y = self.ceilingHeight

        # fall
        if self.rec.y <= self.floorHeight:
            self.yvelocity += 0.68
        if self.rec.y >= self.floorHeight:
            self.yvelocity = 0

        # controls
        if IsKeyDown(KEY_RIGHT):
            self.xvelocity += 0.2
        if IsKeyDown(KEY_LEFT):
            self.xvelocity -= 0.2
        if IsKeyDown(KEY_SPACE):
                if self.frames % 6 == 0 and self.stopIncrementingTickTimer == False:
                    self.tickTimer += 1
                if self.tickTimer < 4:
                    self.yvelocity -= 1.5
                elif self.tickTimer > 5 and self.rec.y > 795:
                    self.yvelocity = 4
                    self.stopIncrementingTickTimer = True
        
        if IsKeyReleased(KEY_RIGHT):
            self.xvelocity = 0
        if IsKeyReleased(KEY_LEFT):
            self.xvelocity = 0

        if IsKeyReleased(KEY_SPACE):
            self.yvelocity = 4 
            self.tickTimer = 0
            self.stopIncrementingTickTimer = False

        # velocity
        if self.xvelocity != 0:
            self.rec.x += self.xvelocity
        if self.yvelocity != 0:
            self.rec.y += self.yvelocity

File: swift-edition/liner-swift-edition/liner-swift-edition/character/player.swift
//
//  player.swift
//  liner-swift-edition
//
//  Created by easontek2398 on 8/6/22.
//

import Foundation
import Raylib

class Player {
    var frames = 0
    var ticks = 0
    
    var rec:Rectangle? = nil
    var xvelocity:Float = 0
    var yvelocity:Float = 0
    var floorHeight: Float = 1000
    var currentFrame = 0
    
    var timer = 0
    
    init(X posx:Float, Y posy:Float, Width width:Float, Height height:Float, Color color:Color) {
        self.rec = Rectangle(x: posx, y: posy, width: width, height: height)
    }
    
    func Physics() {
        if self.rec!.y <= self.floorHeight {
            self.yvelocity += 0.68
        } else {
            self.yvelocity = 0
            self.rec!.y = self.floorHeight
        }
    }
    
    func Controls() {
        if Raylib.isKeyDown(.left) {
            self.xvelocity -= 0.5
        }
        if Raylib.isKeyDown(.right) {
            self.xvelocity += 0.5
        }
        if Raylib.isKeyReleased(.right) || Raylib.isKeyReleased(.left) {
            self.xvelocity = 0
        }
    }
    
    func Update() {
        self.Controls()
        self.Physics()
        self.frames += 1
        if self.frames % 6 == 0 {
            self.ticks += 1
        }
        self.rec!.x += Float(Int(self.xvelocity))
        self.rec!.y += Float(Int(self.yvelocity))
    }
    
    func Draw() {
        Raylib.drawRectangleRec(self.rec!, .gold)
    }
}
